### PR TITLE
Add __type checking

### PR DESCRIPTION
**Pull request by sdf611097**
Add __type checking
Avoid object saved with wrong format from REST API.

### CODE REVIEW

**Comment by gfosco** [comment]
Not sure if we should provide the json-string version of the object in the error messages... I'm thinking we should not.  Do you have an opinion on this @drew-gross ?

**Comment by drew-gross** [comment]
Yeah my suggestion would be to not return that as an error. Too much potential for information to leak to people who shouldn't be able to see it.

**Comment by sdf611097** [comment]
How can I modify these messages, or any variables I can get from JavascriptSDK? The messages are similar as current Parse API server response.

**Comment by sdf611097** [comment]
Hi @gfosco  @drew-gross ,
I modify the error message. Is that what you want?

File: src/Schema.js [hunk 1]
```diff
@@ -731,19 +731,44 @@ function getObjectType(obj) {
...
+          throw new Parse.Error(Parse.Error.INCORRECT_TYPE, "This is not a valid "+obj.__type);
+        }
+        break;
+      case 'Bytes' :
+        if(!obj.base64) {
+          throw new Parse.Error(Parse.Error.INCORRECT_TYPE, "This is not a valid "+obj.__type);
+        }
+        break;
+      default :
+        throw new Parse.Error(Parse.Error.INCORRECT_TYPE, 'invalid type: ' + obj.__type);
```
Comment (flovilmart) [inline src/Schema.js]: as each case returns, you could just add after the switch `throw new Parse.Error(Parse.Error.INCORRECT_TYPE, 'invalid type: ' + obj.__type); `

**Comment by drew-gross** [comment]
@flovilmart has a good suggestion, but I'm going to merge this anyway. If you want to updated in a new PR, that would make the code a little cleaner :)

**Pull request merged**